core/services/src/main/java/org/exoplatform/services/cms/clouddrives/CloudDriveServiceImpl.java

```diff
@@ -580,4 +583,31 @@ protected void loadConnected(ManageableRepository jcrRepository) {
       sessionProviders.setSessionProvider(null, spOrig);
     }
   }
+
+  @Override
+  public void disconnectCloudDrive(String workspace, String path, String providerId) {
+    String userName = getCurrentUserIdentity().getUserId();
+    SessionProvider sp = sessionProviders.getSessionProvider(null);
+    try {
+      Session userSession = sp.getSession(workspace, jcrService.getCurrentRepository());
+      Item item = finder.findItem(userSession, path);
+      Node userNode = (Node) item;
+      String queryStr = "select * from " + JCRLocalCloudDrive.ECD_CLOUDDRIVE + " where ecd:localUserName='" + userName
+          + "' AND ecd:provider='" + providerId + "'";
+      Query q = userNode.getSession().getWorkspace().getQueryManager().createQuery(queryStr, Query.SQL);
+      NodeIterator r = q.execute().getNodes();
+      if (r.hasNext()) {
+        Node driveNode = r.nextNode();
```

Comment:
this will disconnect all accounts related to, the same connector. we need to send the exact email account to be deleted and match it with the corresponding jcr property.